### ISSUE TITLE
Betygsätt gäster

### ISSUE THREAD

**Issue by FannyBacklin** [noQ-sweden/noq#545]
**Beskrivning/Story**
Som en handläggare på socialtjänsten vill jag kunna betygsätta gästerna efter deras vistelse genom att ge 1-5 stjärnor för att säkerställa kvalitétsstandard och trygghet. 

**Acceptance Criteria**

AC1:
* Förutsatt [att gästen har checkat ut]
* När [handläggaren har bekräftat utcheckningen]
* Då [får handläggaren upp valet att betygsätta gästen genom att ange 1-5 stjärnor]

**Subtasks**

###Definition of Done
OK from wingmate
Code checked in
All subtasks done or cancelled
Max 5 min demo video

**Comment by FannyBacklin** [comment]
@LudvigErlandsson  funderar kring omformulering / ta bort antalet stjärnor.

**Comment by LudvigErlandsson** [comment]
Vi pausar denna story tills vidare. Den kommer bakas in i Riskbedömnings-epicen istället, så utveckling lär dra igång i januari istället!

**Comment by villenoqnu** [comment]
@AbubakerSalah @rohan-noQ

**Comment by patrik-pwned** [comment]
So, this ticket was originally intended for guests at shelters, as you can guess in Jani's comment with new AC. For the moment and for the sake of time, I'd say however to limit yourselves to the ticket's stated AC at the top @AbubakerSalah @rohan-noQ 

We also need to research further on the needs of Help Organizations's quality assurance needs and Volunteers' gamification before putting more effort but if you have good ideas then let me know :)

**Comment by patrik-pwned** [comment]
The volunteer. 
Would it be too much to add feature to Guest as well? If so, we'll create new ticket later so no problem :)

**Comment by AbubakerSalah** [comment]
We will do for the volunteers for now and see how it goes :)

**Comment by AbubakerSalah** [comment]
Hej, @patrik-pwned @FannyBacklin @jani-at-noq ,

We’re having an issue with the backend implementation. Can you clarify if the volunteer rating should be based on how many guests they’ve booked or on how many assignments they’ve completed?

**Comment by WrenHuan9** [comment]
Yes, in fact, there are quite a few unclear aspects regarding this scenario.

1. If the rating is meant for the volunteer, is there a specific object associated with the score? For example, if the volunteer helps a guest send a booking, should the host be the one rating the volunteer's performance during that process? Or is the rating meant to happen when the volunteer takes an assignment from the host and completes it, at which point feedback on the volunteer's performance is provided?

2. If the rating is targeting the volunteer's action of booking on behalf of the guest, the current submitted booking request does not contain any reference to the volunteer's identity. We only know who the final guest is (based on the latest version of the data model documentation), but not which volunteer submitted the booking request.

3. regarding the "take assignment" action, it seems that in the current flow, there is no confirmation step from the host to close the assignment. So what is the trigger condition for the feedback? If feedback can be given at any point during the execution of the assignment, it might lead to the same host rating the same volunteer multiple times (or even maliciously). Has this been considered?

Given the above, we think at this stage, implementing the rating feature for guests might be more reasonable, especially since the acceptance criteria mention triggering the rating after clicking "checkout" button. What do you think about it or could it be that our understanding of the scenario is flawed in some way?

**Comment by patrik-pwned** [comment]
@WrenHuan9 
1. For the future, we can consider a rating feature for the guests/help org regarding bookings. I think it would make sense to allow guests/help orgs rate a volunteer on any performance BUT **for the scope of this ticket, it will suffice to allow the help org to rate volunteers who have signed up for an assignment AND the assignment has begun**. So, the rating can only take place on volunteers who have registered for an assignment that has taken place.
(If we want to think about a future scenario, we can imagine that the help org can only rate a volunteer that has participated but if the volunteer signs up and doesn't show up then this could be aggregated into the volunteer's profile performance record as "n times didn't show up" for example.)
2. All right, that's something we need to consider for the future. Good point!
3. It has been not considered but it would be make sense for the trigger to take place when the assignment ends in the app.

**Comment by FannyBacklin** [comment]
Very good feedback @AbubakerSalah and @WrenHuan9! 🙌

@patrik-pwned – if I may play devil’s advocate here (hehe 😄) – have we checked in with the help orgs and volunteers (ping @overath2) if this is something that would actually bring value?

For example, if a volunteer has a 3 out of 5 star rating – how would that information be used? Would the help org act on that in any way? Or could it risk being an extra step in the process without any real impact?

Also, how are these kinds of situations handled today? Might be helpful to understand the current workflow before we build something new.

Just thinking out loud here :)

**Comment by patrik-pwned** [comment]
Good question!

In my initial discussion with the help org, it became clear that social proof or recognition of performance is a crucial motivator for volunteers - whether to themselves or to show their peers. Making ambassadors out of the volunteers will also be a big help to market the product through word-of-mouth. That's why the teams concluded that it should be part of an MVP. 

The rating system is a stepping stone to the bigger gamification [spike] epic that we're planning. Within this epic, I can say that we need to explore ideas further on how expand this completely so all thoughts are welcome :)

**Comment by jani-at-noq** [comment]
Would it not be better to have badges for number of times person has been a volunteer instead of a rating? For 1st time, 10th time, 50th time etc. If I would get rating 1 star I would not go back.

**Comment by patrik-pwned** [comment]
Possibly. I think this topic is suited well for the spike we have on gamification. There are no big ideas written in stone yet
